app/models/reviewer_access.rb

```diff
@@ -0,0 +1,69 @@
+# manage anonymous access to private studies via access code & pin
+# will generate a short-lived session for reviewers with valid credentials
+class ReviewerAccess
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  field :access_code, type: String # UUID
+  field :pin, type: String # 6 character string
```

Comment:
We should increase the number of characters (and possibly their type). 6-pin is considered quite "trivial" from brute-forcing standpoint. You may want to introduce other alphanumeric characters to increase the entropy too.

Reply:
Sure, I can have it generate from any alphanumeric.

Reply:
Also, are the PIN characters submitted by the user validated somehow? I.e. is it just an "equality" comparison or do you restrict the character set on the backend also?

Reply:
It's just an equality comparison, but I like where you're going with that.  I should strip non-alphanumeric characters.  The form on the front end limits the characters, so a user can't enter more that the allowed limit of 6.  I could also do the same on the backend as well, just to be sure.

Reply:
Thanks! The main concern here is we should use a "secure equality" operator, since this is a common DoS vulnerability when a regular `==` is used. So perhaps take a look at https://github.com/ConradIrwin/secure_equals or something similar.

Reply:
Gotcha.  I found this which looks pretty robust - it's part of `ActiveSupport` which is a core Rails library:

https://api.rubyonrails.org/classes/ActiveSupport/SecurityUtils.html

Reply:
Awesome, this seems great!